packages/@tinacms/cli/src/cmds/compile/index.ts

```diff
@@ -313,15 +297,49 @@ export const compileSchema = async (options: {
   schemaFileType?: string
   verbose?: boolean
   dev?: boolean
-  rootPath
+  rootPath: string
 }) => {
   const root = options.rootPath
   const tinaPath = path.join(root, '.tina')
   const tinaGeneratedPath = path.join(tinaPath, '__generated__')
   const tinaConfigPath = path.join(tinaGeneratedPath, 'config')
 
-  let schema: any = await compileFile(options, 'schema')
-  try {
+  let schemaExists = fileExists({
+    projectDir: tinaPath,
+    filename: 'schema',
+    allowedTypes: ['js', 'jsx', 'tsx', 'ts'],
+  })
+  const configExists = fileExists({
+    projectDir: tinaPath,
+    filename: 'config',
+    allowedTypes: ['js', 'jsx', 'tsx', 'ts'],
+  })
+
+  // If there is not Schema and no config
+  if (!schemaExists && !configExists) {
```

Comment:
For what it's worth I'm not sure there's a ton of value in creating the schema at this stage. I know this command is run for various tasks, but I think the main path it'd be preferred to just throw an error saying you need to run `tinacms init`.

Reply:
Agreed. Pushed up a fix for that.